NEWS.md

```diff
@@ -3,6 +3,306 @@ XGBoost Change Log
+* Fix DART prediction in R (#5204)
+
+### Make external memory more robust
+* Fix issues with training with external memory on cpu (#4487)
+* Fix crash with approx tree method on cpu (#4510)
+* Fix external memory race in `exact` (#4980). Note: `dmlc::ThreadedIter` is not actually thread-safe. We would like to re-design it in the long term.
+
+### Major refactoring of the `DMatrix` class (#4686, #4744, #4748, #5044, #5092, #5108, #5188, #5198)
+* Goal 1: improve performance and reduce memory consumption. Right now, if the user trains a model with a NumPy array as training data, the array gets copies 2-3 times before training begins. We'd like to reduce duplication of the data matrix.
+* Goal 2: Expose a common interface to external data, unify the way DMatrix objects are constructed and simplify the process of adding new external data sources. This work is essential for ingesting cuPy arrays.
+* RFC: #4354, Roadmap: #5143
+* This work is also relevant to external memory support on GPUs.
+
+### Breaking: XGBoost Python package now requires Python 3.5 or newer (#5021, #5274)
+* Python 3.4 has reached its end-of-life on March 16, 2019, so we now require Python 3.5 or newer.
+
+### Breaking: GPU algorithm now requires CUDA 9.0 and higher (#4527, #4580)
+
+### Breaking: `n_gpus` parameter removed; multi-GPU training now requires a distributed framework (#4579, #4749, #4773, #4810, #4867, #4908)
+* #4531 proposed removing support for single-process multi-GPU training. Contributors would focus on multi-GPU support through distributed frameworks such as Dask and Spark, where the framework would be expected to assign a worker process for each GPU independently. By delegating GPU management and data movement to the distributed framework, we can greatly simplify the core XGBoost codebase, make multi-GPU training more robust, and reduce burden for future development.
+
+### Breaking: Some deprecated features have been removed
+* ``gpu_exact`` training method (#4527, #4742, #4777). Use ``gpu_hist`` instead.
+* ``learning_rates`` parameter in Python (#5155). Use the callback API instead.
+* ``num_roots`` (#5059, #5165), since the current training code always uses a single root node.
+* GPU-specific objectives (#4690), such as `gpu:reg:linear`. Use objectives without `gpu:` prefix; GPU will be used automatically if your machine has one.
+
+### Performance improvements
+* Smarter choice of histogram construction for distributed `gpu_hist` (#4519)
+* Optimizations for quantization on device (#4572)
+* Introduce caching memory allocator to avoid latency associated with GPU memory allocation (#4554, #4615)
+* Optimize the initialization stage of the CPU `hist` algorithm for sparse datasets (#4625)
+* Prevent unnecessary data copies from GPU memory to the host (#4795)
+* Improve operation efficiency for single prediction (#5016)
+* Group builder modified for incremental building, to speed up building large `DMatrix` (#5098)
+
+### Bug-fixes
```

Comment:
In #4990 (run training on empty DMatix), I also enforced each XGBoost process can only run on different GPUs to avoid nccl error.